### ISSUE TITLE
Optimizer params groups split 

### ISSUE THREAD

**Issue by FraPochetti** [ultralytics/yolov5#2422]
## ❔Question
In [here](https://github.com/ultralytics/yolov5/blob/e8a2b83268950e346899a84e8d29e84d178553b1/train.py#L101) I see that you are splitting parameters according to them being `biases`, from `BN` layers, or all the rest.

It is the first time I stumble upon this kind of approach. I generally see models split in chunks along their depth, to allow adopting lower learning rates for deeper layers and higher ones as we approach the head.

Can you please elaborate on this choice?
Thanks

**Comment by glenn-jocher** [comment]
@FraPochetti BN layers are separated into an independent parameter group to remove weight_decay from applying to them.

Bias layers are added to an independent parameter group to apply higher learning rate during warmup (mainly benefits training from scratch, less so finetuning).

The depth approach is interesting, I've not experimented with this in YOLOv5. What sort of LR ratios might be used from the first to last layer?

If you come up with any interesting results using this method please let us know!

**Comment by FraPochetti** [comment]
@glenn-jocher thanks for the prompt reply!
This technique is called Discriminative Learning Rates and it was introduced by the fastai folks.
You can read more about it [here](https://github.com/fastai/fastbook/blob/master/05_pet_breeds.ipynb) (search for `Discriminative Learning Rates`).
It is mainly beneficial for fine-tuning, to be precise.
Jeremy Howard suggests using a ratio of 100, in general. E.g. the head is trained at `lr`, the deepest param group at `lr/100` and the groups in between at equidistant rates between min and max.
It works pretty well in practice so I might try this out and report back!

**Comment by glenn-jocher** [comment]
@FraPochetti wow that's a significant difference. We have two sets of hyps BTW in data/ you should be aware of, and sometimes finetuning on hyp.scratch.yaml will lead to very significant early overfitting, particularly in the BCELoss terms obj and cls. I wonder if this technique might reduce that.

- hyp.scratch.yaml: evolved on COCO training from scratch
- hyp.finetune.yaml: evolved on VOC finetuning from COCO pretrained models

**Comment by FraPochetti** [comment]
@glenn-jocher It is indeed quite a difference.
I will try to experiment with finetuning and Discriminative Learning Rates (DLR) using the `hyp.finetune.yaml` hparams plus the pretrained weights you guys offer.
Not sure DLR helps with training from scratch but we might give it a go there too.
Will keep you posted!
